Comparing metadata for images that don't exist locally causes panic due to SIGSEGV

## Observed

When running `container-diff` using one or more images that do not exist locally and passing `--type=metadata`, there is a panic :O

```bash
container-diff diff cats dogs --type=metadata 
2018/07/25 15:31:31 No matching credentials found for index.docker.io, falling back on anonymous
2018/07/25 15:31:31 No matching credentials found for index.docker.io, falling back on anonymous
WARN[0001] Diff may be inaccurate: open : no such file or directory 
WARN[0001] Diff may be inaccurate: open : no such file or directory 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x81f612]

goroutine 1 [running]:
github.com/GoogleContainerTools/container-diff/differs.getMetadataList(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/GoogleContainerTools/container-diff/differs/metadata_diff.go:77 +0x42
github.com/GoogleContainerTools/container-diff/differs.getMetadataDiff(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/GoogleContainerTools/container-diff/differs/metadata_diff.go:62 +0x8a
github.com/GoogleContainerTools/container-diff/differs.MetadataAnalyzer.Diff(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/GoogleContainerTools/container-diff/differs/metadata_diff.go:40 +0xb2
github.com/GoogleContainerTools/container-diff/differs.(*MetadataAnalyzer).Diff(0xc7ff18, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	<autogenerated>:1 +0xab
github.com/GoogleContainerTools/container-diff/differs.DiffRequest.GetDiff(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/GoogleContainerTools/container-diff/differs/differs.go:62 +0x1b8
github.com/GoogleContainerTools/container-diff/cmd.diffImages(0x7ffd24b21fc0, 0x4, 0x7ffd24b21fc5, 0x4, 0xc42028fce0, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/container-diff/cmd/diff.go:110 +0x601
github.com/GoogleContainerTools/container-diff/cmd.glob..func4(0xc57a60, 0xc4202f4330, 0x2, 0x3)
	/go/src/github.com/GoogleContainerTools/container-diff/cmd/diff.go:46 +0x89
github.com/GoogleContainerTools/container-diff/vendor/github.com/spf13/cobra.(*Command).execute(0xc57a60, 0xc4202f42a0, 0x3, 0x3, 0xc57a60, 0xc4202f42a0)
	/go/src/github.com/GoogleContainerTools/container-diff/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/GoogleContainerTools/container-diff/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc57cc0, 0x0, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/container-diff/vendor/github.com/spf13/cobra/command.go:852 +0x334
github.com/GoogleContainerTools/container-diff/vendor/github.com/spf13/cobra.(*Command).Execute(0xc57cc0, 0x1f, 0x0)
	/go/src/github.com/GoogleContainerTools/container-diff/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
	/go/src/github.com/GoogleContainerTools/container-diff/main.go:35 +0x66
```


## Expected
There should be an error message probably, but not a panic/runtime exception, e.g. without `--type=metadata` the output is:

```bash
➜  kaniko git:(local_integration_tests) container-diff diff cats dogs                
2018/07/25 15:33:08 No matching credentials found for index.docker.io, falling back on anonymous
2018/07/25 15:33:08 No matching credentials found for index.docker.io, falling back on anonymous
WARN[0001] Diff may be inaccurate: open : no such file or directory 
WARN[0001] Diff may be inaccurate: open : no such file or directory 
ERRO[0001] Error getting diff with AptAnalyzer: stat : no such file or directory 
ERRO[0001] Could not retrieve diff: Could not perform diff on {<nil>   []} and {<nil>   []} 
```
Interestingly, the output also doesn't match - but at least doesn't panic! - when using `--type=file`:

```bash
➜  kaniko git:(local_integration_tests) container-diff diff cats dogs --type=file    
2018/07/25 15:33:35 No matching credentials found for index.docker.io, falling back on anonymous
2018/07/25 15:33:35 No matching credentials found for index.docker.io, falling back on anonymous
WARN[0001] Diff may be inaccurate: open : no such file or directory 
WARN[0001] Diff may be inaccurate: open : no such file or directory 

-----File-----

These entries have been added to : None

These entries have been deleted from : None

These entries have been changed between  and : None
```

## Steps to reproduce
Assuming you do not have the images `cat` and `dog` locally (or run `docker system prune -a` so you can use ANY image name XD), run:

```bash
container-diff diff cats dogs --type=metadata
```

_This is with version `v0.11.0`._

--- a/cmd/analyze.go
+++ b/cmd/analyze.go
@@ -17,13 +17,13 @@ limitations under the License.
 package cmd
 
 import (
-	"errors"
 	"fmt"
 	"os"
 
 	"github.com/GoogleContainerTools/container-diff/cmd/util/output"
 	"github.com/GoogleContainerTools/container-diff/differs"
 	pkgutil "github.com/GoogleContainerTools/container-diff/pkg/util"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
@@ -56,19 +56,19 @@ func checkAnalyzeArgNum(args []string) error {
 func analyzeImage(imageName string, analyzerArgs []string) error {
 	analyzeTypes, err := differs.GetAnalyzers(analyzerArgs)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "getting analyzers")
 	}
 
 	image, err := getImageForName(imageName)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "error retrieving image %s", imageName)
 	}
 
 	if noCache && !save {
 		defer pkgutil.CleanupImage(image)
 	}
 	if err != nil {
-		return fmt.Errorf("Error processing image: %s", err)
+		return fmt.Errorf("error processing image: %s", err)
 	}
 
 	req := differs.SingleRequest{
@@ -76,7 +76,7 @@ func analyzeImage(imageName string, analyzerArgs []string) error {
 		AnalyzeTypes: analyzeTypes}
 	analyses, err := req.GetAnalysis()
 	if err != nil {
-		return fmt.Errorf("Error performing image analysis: %s", err)
+		return fmt.Errorf("error performing image analysis: %s", err)
 	}
 
 	logrus.Info("retrieving analyses")

--- a/cmd/analyze_test.go
+++ b/cmd/analyze_test.go
@@ -29,11 +29,11 @@ var analyzeArgNumTests = []testpair{
 func TestAnalyzeArgNum(t *testing.T) {
 	for _, test := range analyzeArgNumTests {
 		err := checkAnalyzeArgNum(test.input)
-		if (err == nil) != test.expected_output {
-			if test.expected_output {
-				t.Errorf("Got unexpected error: %s", err)
-			} else {
+		if (err == nil) != test.shouldError {
+			if test.shouldError {
 				t.Errorf("Expected error but got none")
+			} else {
+				t.Errorf("Got unexpected error: %s", err)
 			}
 		}
 	}

--- a/cmd/diff.go
+++ b/cmd/diff.go
@@ -79,6 +79,24 @@ func processImage(imageName string, imageMap map[string]*pkgutil.Image, wg *sync
 	imageMap[imageName] = &image
 }
 
+// collects errors from a channel and combines them
+// assumes channel has already been closed
+func readErrorsFromChannel(c chan error) error {
+	errs := []string{}
+	for {
+		err, ok := <-c
+		if !ok {
+			break
+		}
+		errs = append(errs, err.Error())
+	}
+
+	if len(errs) > 0 {
+		return errors.New(strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
 func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 	diffTypes, err := differs.GetAnalyzers(diffArgs)
 	if err != nil {
@@ -104,17 +122,8 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 		defer pkgutil.CleanupImage(*imageMap[image2Arg])
 	}
 
-	errs := []string{}
-	for {
-		err, ok := <-errChan
-		if !ok {
-			break
-		}
-		errs = append(errs, err.Error())
-	}
-
-	if len(errs) > 0 {
-		return errors.New(strings.Join(errs, "\n"))
+	if err := readErrorsFromChannel(errChan); err != nil {
+		return err
 	}
 
 	logrus.Info("computing diffs")

--- a/cmd/diff.go
+++ b/cmd/diff.go
@@ -17,7 +17,6 @@ limitations under the License.
 package cmd
 
 import (
-	"errors"
 	"fmt"
 	"os"
 	"sync"
@@ -26,6 +25,7 @@ import (
 	"github.com/GoogleContainerTools/container-diff/differs"
 	pkgutil "github.com/GoogleContainerTools/container-diff/pkg/util"
 	"github.com/GoogleContainerTools/container-diff/util"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
@@ -69,10 +69,23 @@ func checkFilenameFlag(_ []string) error {
 	return errors.New("Please include --types=file with the --filename flag")
 }
 
+func processImage(imageName string, imageMap map[string]*pkgutil.Image, wg *sync.WaitGroup, errChan chan<- error) {
+	defer wg.Done()
+	image, err := getImageForName(imageName)
+	if image.Image == nil {
+		errChan <- fmt.Errorf("error retrieving image %s: %s", imageName, err.Error())
+		return
+	}
+	if err != nil {
+		logrus.Warningf("diff may be inaccurate: %s", err)
+	}
+	imageMap[imageName] = &image
+}
+
 func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 	diffTypes, err := differs.GetAnalyzers(diffArgs)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "getting analyzers")
 	}
 
 	var wg sync.WaitGroup
@@ -80,22 +93,34 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 
 	logrus.Infof("starting diff on images %s and %s, using differs: %s\n", image1Arg, image2Arg, diffArgs)
 
-	imageMap := map[string]*pkgutil.Image{
-		image1Arg: {},
-		image2Arg: {},
-	}
-	// TODO: fix error handling here
-	for imageArg := range imageMap {
-		go func(imageName string, imageMap map[string]*pkgutil.Image) {
-			defer wg.Done()
-			image, err := getImageForName(imageName)
-			imageMap[imageName] = &image
-			if err != nil {
-				logrus.Warningf("Diff may be inaccurate: %s", err)
-			}
-		}(imageArg, imageMap)
-	}
+	imageMap := map[string]*pkgutil.Image{}
+	errChan := make(chan error, 2)
+
+	go processImage(image1Arg, imageMap, &wg, errChan)
+	go processImage(image2Arg, imageMap, &wg, errChan)
+
 	wg.Wait()
+	err, ok := <-errChan
+	errs := []error{}
+	if ok {
+		errs = append(errs, err)
+	}
+	if len(errs) > 0 {
+		errStr := ""
+		for _, err := range errs {
+			errStr = errStr + err.Error()
+		}
+		return errors.New(errStr)
+	}
+
+	img1, ok := imageMap[image1Arg]
+	if !ok {
+		return fmt.Errorf("cannot find image %s", image1Arg)
+	}
+	img2, ok := imageMap[image2Arg]
+	if !ok {
+		return fmt.Errorf("cannot find image %s", image2Arg)
+	}
 
 	if noCache && !save {
 		defer pkgutil.CleanupImage(*imageMap[image1Arg])
@@ -104,8 +129,8 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 
 	logrus.Info("computing diffs")
 	req := differs.DiffRequest{
-		Image1:    *imageMap[image1Arg],
-		Image2:    *imageMap[image2Arg],
+		Image1:    *img1,
+		Image2:    *img2,
 		DiffTypes: diffTypes}
 	diffs, err := req.GetDiff()
 	if err != nil {

--- a/cmd/diff.go
+++ b/cmd/diff.go
@@ -19,6 +19,7 @@ package cmd
 import (
 	"fmt"
 	"os"
+	"strings"
 	"sync"
 
 	"github.com/GoogleContainerTools/container-diff/cmd/util/output"
@@ -100,17 +101,19 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 	go processImage(image2Arg, imageMap, &wg, errChan)
 
 	wg.Wait()
-	err, ok := <-errChan
-	errs := []error{}
-	if ok {
-		errs = append(errs, err)
-	}
-	if len(errs) > 0 {
-		errStr := ""
-		for _, err := range errs {
-			errStr = errStr + err.Error()
+	close(errChan)
+
+	errs := []string{}
+	for {
+		err, ok := <-errChan
+		if !ok {
+			break
 		}
-		return errors.New(errStr)
+		errs = append(errs, err.Error())
+	}
+
+	if len(errs) > 0 {
+		return errors.New(strings.Join(errs, "\n"))
 	}
 
 	img1, ok := imageMap[image1Arg]

--- a/cmd/diff.go
+++ b/cmd/diff.go
@@ -67,18 +67,14 @@ func checkFilenameFlag(_ []string) error {
 			return nil
 		}
 	}
-	return errors.New("Please include --types=file with the --filename flag")
+	return errors.New("please include --types=file with the --filename flag")
 }
 
 func processImage(imageName string, imageMap map[string]*pkgutil.Image, wg *sync.WaitGroup, errChan chan<- error) {
 	defer wg.Done()
 	image, err := getImageForName(imageName)
-	if image.Image == nil {
-		errChan <- fmt.Errorf("error retrieving image %s: %s", imageName, err.Error())
-		return
-	}
 	if err != nil {
-		logrus.Warningf("diff may be inaccurate: %s", err)
+		errChan <- fmt.Errorf("error retrieving image %s: %s", imageName, err)
 	}
 	imageMap[imageName] = &image
 }
@@ -103,6 +99,11 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 	wg.Wait()
 	close(errChan)
 
+	if noCache && !save {
+		defer pkgutil.CleanupImage(*imageMap[image1Arg])
+		defer pkgutil.CleanupImage(*imageMap[image2Arg])
+	}
+
 	errs := []string{}
 	for {
 		err, ok := <-errChan
@@ -116,28 +117,14 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 		return errors.New(strings.Join(errs, "\n"))
 	}
 
-	img1, ok := imageMap[image1Arg]
-	if !ok {
-		return fmt.Errorf("cannot find image %s", image1Arg)
-	}
-	img2, ok := imageMap[image2Arg]
-	if !ok {
-		return fmt.Errorf("cannot find image %s", image2Arg)
-	}
-
-	if noCache && !save {
-		defer pkgutil.CleanupImage(*imageMap[image1Arg])
-		defer pkgutil.CleanupImage(*imageMap[image2Arg])
-	}
-
 	logrus.Info("computing diffs")
 	req := differs.DiffRequest{
-		Image1:    *img1,
-		Image2:    *img2,
+		Image1:    *imageMap[image1Arg],
+		Image2:    *imageMap[image2Arg],
 		DiffTypes: diffTypes}
 	diffs, err := req.GetDiff()
 	if err != nil {
-		return fmt.Errorf("Could not retrieve diff: %s", err)
+		return fmt.Errorf("could not retrieve diff: %s", err)
 	}
 	outputResults(diffs)
 
@@ -150,7 +137,7 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 	}
 
 	if noCache && save {
-		logrus.Infof("Images were saved at %s and %s", imageMap[image1Arg].FSPath,
+		logrus.Infof("images were saved at %s and %s", imageMap[image1Arg].FSPath,
 			imageMap[image2Arg].FSPath)
 	}
 	return nil

--- a/cmd/diff.go
+++ b/cmd/diff.go
@@ -70,6 +70,7 @@ func checkFilenameFlag(_ []string) error {
 	return errors.New("please include --types=file with the --filename flag")
 }
 
+// processImage is a concurrency-friendly wrapper around getImageForName
 func processImage(imageName string, imageMap map[string]*pkgutil.Image, wg *sync.WaitGroup, errChan chan<- error) {
 	defer wg.Done()
 	image, err := getImageForName(imageName)

--- a/cmd/diff_test.go
+++ b/cmd/diff_test.go
@@ -21,21 +21,46 @@ import (
 )
 
 var diffArgNumTests = []testpair{
-	{[]string{}, false},
-	{[]string{"one"}, false},
-	{[]string{"one", "two"}, true},
-	{[]string{"one", "two", "three"}, false},
+	{[]string{}, true},
+	{[]string{"one"}, true},
+	{[]string{"one", "two"}, false},
+	{[]string{"one", "two", "three"}, true},
 }
 
 func TestDiffArgNum(t *testing.T) {
 	for _, test := range diffArgNumTests {
 		err := checkDiffArgNum(test.input)
-		if (err == nil) != test.expected_output {
-			if test.expected_output {
-				t.Errorf("Got unexpected error: %s", err)
-			} else {
-				t.Errorf("Expected error but got none")
-			}
+		checkError(t, err, test.shouldError)
+	}
+}
+
+type imageDiff struct {
+	image1      string
+	image2      string
+	shouldError bool
+}
+
+var imageDiffs = []imageDiff{
+	{"", "", true},
+	{"gcr.io/google-appengine/python", "gcr.io/google-appengine/debian9", false},
+	{"gcr.io/google-appengine/python", "cats", true},
+}
+
+func TestDiffImages(t *testing.T) {
+	for _, test := range imageDiffs {
+		err := diffImages(test.image1, test.image2, []string{"apt"})
+		checkError(t, err, test.shouldError)
+		err = diffImages(test.image1, test.image2, []string{"metadata"})
+		checkError(t, err, test.shouldError)
+	}
+}
+
+func checkError(t *testing.T, err error, shouldError bool) {
+	if (err == nil) == shouldError {
+		if shouldError {
+			t.Errorf("expected error but got none")
+		} else {
+			t.Errorf("got unexpected error: %s", err)
 		}
 	}
 }

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -132,6 +132,9 @@ func checkIfValidAnalyzer(_ []string) error {
 	return nil
 }
 
+// getImageForName infers the source of an image and retrieves a v1.Image reference to it.
+// Once a reference is obtained, it attempts to unpack the v1.Image's reader's contents
+// into a temp directory on the local filesystem.
 func getImageForName(imageName string) (pkgutil.Image, error) {
 	logrus.Infof("retrieving image: %s", imageName)
 	var img v1.Image

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -38,6 +38,7 @@ import (
 	"github.com/GoogleContainerTools/container-diff/util"
 	"github.com/google/go-containerregistry/pkg/v1"
 	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
@@ -139,17 +140,17 @@ func getImageForName(imageName string) (pkgutil.Image, error) {
 		start := time.Now()
 		img, err = tarball.ImageFromPath(imageName, nil)
 		if err != nil {
-			return pkgutil.Image{}, err
+			return pkgutil.Image{}, errors.Wrap(err, "retrieving tar from path")
 		}
 		elapsed := time.Now().Sub(start)
-		logrus.Infof("retrieving image from tar took %f seconds", elapsed.Seconds())
+		logrus.Infof("retrieving image ref from tar took %f seconds", elapsed.Seconds())
 	} else if strings.HasPrefix(imageName, DaemonPrefix) {
 		// remove the daemon prefix
 		imageName = strings.Replace(imageName, DaemonPrefix, "", -1)
 
 		ref, err := name.ParseReference(imageName, name.WeakValidation)
 		if err != nil {
-			return pkgutil.Image{}, err
+			return pkgutil.Image{}, errors.Wrap(err, "parsing image reference")
 		}
 
 		start := time.Now()
@@ -158,28 +159,28 @@ func getImageForName(imageName string) (pkgutil.Image, error) {
 			Buffer: true,
 		})
 		if err != nil {
-			return pkgutil.Image{}, err
+			return pkgutil.Image{}, errors.Wrap(err, "retrieving image from daemon")
 		}
 		elapsed := time.Now().Sub(start)
-		logrus.Infof("retrieving image from daemon took %f seconds", elapsed.Seconds())
+		logrus.Infof("retrieving local image ref took %f seconds", elapsed.Seconds())
 	} else {
 		// either has remote prefix or has no prefix, in which case we force remote
 		imageName = strings.Replace(imageName, RemotePrefix, "", -1)
 		ref, err := name.ParseReference(imageName, name.WeakValidation)
 		if err != nil {
-			return pkgutil.Image{}, err
+			return pkgutil.Image{}, errors.Wrap(err, "parsing image reference")
 		}
 		auth, err := authn.DefaultKeychain.Resolve(ref.Context().Registry)
 		if err != nil {
-			return pkgutil.Image{}, err
+			return pkgutil.Image{}, errors.Wrap(err, "resolving auth")
 		}
 		start := time.Now()
 		img, err = remote.Image(ref, auth, http.DefaultTransport)
 		if err != nil {
-			return pkgutil.Image{}, err
+			return pkgutil.Image{}, errors.Wrap(err, "retrieving remote image")
 		}
 		elapsed := time.Now().Sub(start)
-		logrus.Infof("retrieving remote image took %f seconds", elapsed.Seconds())
+		logrus.Infof("retrieving remote image ref took %f seconds", elapsed.Seconds())
 	}
 
 	// create tempdir and extract fs into it
@@ -188,7 +189,7 @@ func getImageForName(imageName string) (pkgutil.Image, error) {
 		start := time.Now()
 		imgLayers, err := img.Layers()
 		if err != nil {
-			return pkgutil.Image{}, err
+			return pkgutil.Image{}, errors.Wrap(err, "getting image layers")
 		}
 		for _, layer := range imgLayers {
 			layerStart := time.Now()
@@ -197,12 +198,12 @@ func getImageForName(imageName string) (pkgutil.Image, error) {
 			if err != nil {
 				return pkgutil.Image{
 					Layers: layers,
-				}, err
+				}, errors.Wrap(err, "getting extract path for layer")
 			}
 			if err := pkgutil.GetFileSystemForLayer(layer, path, nil); err != nil {
 				return pkgutil.Image{
 					Layers: layers,
-				}, err
+				}, errors.Wrap(err, "getting filesystem for layer")
 			}
 			layers = append(layers, pkgutil.Layer{
 				FSPath: path,
@@ -215,12 +216,15 @@ func getImageForName(imageName string) (pkgutil.Image, error) {
 	}
 
 	path, err := getExtractPathForImage(imageName, img)
+	if err != nil {
+		return pkgutil.Image{}, err
+	}
 	// extract fs into provided dir
 	if err := pkgutil.GetFileSystemForImage(img, path, nil); err != nil {
 		return pkgutil.Image{
 			FSPath: path,
 			Layers: layers,
-		}, err
+		}, errors.Wrap(err, "getting filesystem for image")
 	}
 	return pkgutil.Image{
 		Image:  img,

--- a/cmd/root_test.go
+++ b/cmd/root_test.go
@@ -17,6 +17,6 @@ limitations under the License.
 package cmd
 
 type testpair struct {
-	input           []string
-	expected_output bool
+	input       []string
+	shouldError bool
 }

--- a/differs/differs.go
+++ b/differs/differs.go
@@ -63,13 +63,13 @@ func (req DiffRequest) GetDiff() (map[string]util.Result, error) {
 		if diff, err := differ.Diff(img1, img2); err == nil {
 			results[differ.Name()] = diff
 		} else {
-			logrus.Errorf("Error getting diff with %s: %s", differ.Name(), err)
+			logrus.Errorf("error getting diff with %s: %s", differ.Name(), err)
 		}
 	}
 
 	var err error
 	if len(results) == 0 {
-		err = fmt.Errorf("Could not perform diff on %v and %v", img1, img2)
+		err = fmt.Errorf("could not perform diff on %v and %v", img1, img2)
 	} else {
 		err = nil
 	}
@@ -87,13 +87,13 @@ func (req SingleRequest) GetAnalysis() (map[string]util.Result, error) {
 		if analysis, err := analyzer.Analyze(img); err == nil {
 			results[analyzeName] = analysis
 		} else {
-			logrus.Errorf("Error getting analysis with %s: %s", analyzeName, err)
+			logrus.Errorf("error getting analysis with %s: %s", analyzeName, err)
 		}
 	}
 
 	var err error
 	if len(results) == 0 {
-		err = fmt.Errorf("Could not perform analysis on %v", img)
+		err = fmt.Errorf("could not perform analysis on %v", img)
 	} else {
 		err = nil
 	}
@@ -107,11 +107,11 @@ func GetAnalyzers(analyzeNames []string) ([]Analyzer, error) {
 		if a, exists := Analyzers[name]; exists {
 			analyzeFuncs = append(analyzeFuncs, a)
 		} else {
-			return nil, fmt.Errorf("Unknown analyzer/differ specified: %s", name)
+			return nil, fmt.Errorf("unknown analyzer/differ specified: %s", name)
 		}
 	}
 	if len(analyzeFuncs) == 0 {
-		return nil, fmt.Errorf("No known analyzers/differs specified")
+		return nil, fmt.Errorf("no known analyzers/differs specified")
 	}
 	return analyzeFuncs, nil
 }

